### PR TITLE
APPS-363 Implement WDL manifest format

### PR DESCRIPTION
**Pull request by jdidion**
This is an implementation of the manifest format for WDL.

Also updates wdlTools to 0.11.16 and cwlScala to 0.3.0.

### CODE REVIEW

**Comment by commandlinegirl** [comment]
@jdidion Which JIRA ticket is it with?

**Comment by jdidion** [comment]
APPS-363

**Comment by jdidion** [comment]
@commandlinegirl this PR is now complete. Please review.

**Comment by commandlinegirl** [comment]
Does this commit contain anything else than the manifest implementation (so I know what I'm reading)? I can see commits related to empty arrays but after merging with main these changes are not included any more here?

**Comment by jdidion** [comment]
Also updates wdlTools to 0.11.16 and cwlScala to 0.3.0.

**Review by commandlinegirl — COMMENTED** [review]
Minor comments

File: core/src/main/scala/dx/core/ir/Manifest.scala [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+case class Manifest(id: String,
+                    values: Map[String, Value],
+                    types: Map[String, Type],
+                    definitions: Map[String, Type]) {
+  values.keySet.diff(types.keySet) match {
+    case d if d.nonEmpty =>
+      throw new Exception(s"missing type definition(s) ${d.mkString(",")}")
+    case _ => ()
+  }
```
[the remark this answers is not in the document]
Comment (jdidion) [reply core/src/main/scala/dx/core/ir/Manifest.scala:30]: It is now optional.

File: core/src/main/scala/dx/core/ir/ValueSerde.scala [hunk 2]
```diff
@@ -4,34 +4,48 @@ import dx.core.ir.Type._
 import dx.core.ir.Value._
 import spray.json._
 
+import scala.collection.immutable.TreeSeqMap
+
 object ValueSerde extends DefaultJsonProtocol {
+
+  /**
+    * Transform a Value to another Value.
```
[the remark this answers is not in the document]
Comment (jdidion) [reply core/src/main/scala/dx/core/ir/ValueSerde.scala]: done

**Comment by jdidion** [comment]
The comments are resolved. Please re-review @commandlinegirl .

**Review by commandlinegirl — APPROVED** [review]
Please run intergration tests before merging, thanks!

**Pull request merged**